src/Aspire.Hosting/Dashboard/DashboardViewModelService.cs

```diff
@@ -90,6 +91,90 @@ await foreach (var tuple in _kubernetesService.WatchAsync<T>(cancellationToken:
+    /// </summary>
+    private void RunServiceWatchTask()
+    {
+        List<Task> tasks = new List<Task>();
+        _ = Task.Run(async () =>
+        {
+            try
+            {
+                var checkedServices = new HashSet<string>();
+                var maxRetryDuration = TimeSpan.FromSeconds(5);
+                await foreach (var tuple in _kubernetesService.WatchAsync<Service>(cancellationToken: _cancellationToken))
+                {
+                    Task checkTask = Task.Run(async () =>
+                    {
+                        var service = tuple.Item2;
+                        if (service.UsesHttpProtocol(out var uriScheme)
+                            && !checkedServices.Contains(service.Metadata.Name))
+                        {
+                            // We need to verify if service is "actually" working
+                            if (!string.Equals(service.Status?.State, "Ready", StringComparison.OrdinalIgnoreCase))
+                            {
+                                // If service is not ready, we don't try to connect. We will get future notification that service moved to ready state
+                                return;
+                            }
+
+                            var currentTimestamp = DateTime.UtcNow;
+                            var delay = TimeSpan.FromMilliseconds(40);
+                            var errorContent = "404 page not found\n";
+                            var proxyUrlString = $"{uriScheme}://{service.AllocatedAddress}:{service.AllocatedPort}";
+                            while (true)
+                            {
+                                using HttpClient client = new();
+                                try
+                                {
+                                    var response = await client.GetAsync(proxyUrlString).ConfigureAwait(false);
+                                    // Traefik returns a 404 with content length of 19 when it is not ready 
+                                    if (response.StatusCode == HttpStatusCode.NotFound
+                                        && response.Content.Headers.ContentLength == 19)
+                                    {
+                                        var content = await response.Content.ReadAsStringAsync(_cancellationToken).ConfigureAwait(false);
+                                        if (!string.Equals(content, errorContent, StringComparison.Ordinal))
+                                        {
+                                            break;
+                                        }
+                                    }
+                                    else
+                                    {
+                                        break;
+                                    }
+                                }
+                                catch
```

Comment:
Do we want to log the exxceptions for visibility into what failed? For example response is null vs GetAsync failures?

Reply:
Depending on timing, we may still see one or more exceptions while waiting for traefik to warmup as it takes some amount of time for it to start listening on a socket after the process starts. If we wanted to be smarter, we could treat exceptions before we get an http response back different than one before the endpoint responds, but for now having the check timeout after 5 seconds seems a reasonable fallback.

Long term, we'll need to fix the proxy issues causing this workaround to be necessary at all.